builtins/amp-img.js

```diff
@@ -279,6 +279,16 @@ export class AmpImg extends BaseElement {
       this.unlistenLoad_();
       this.unlistenLoad_ = null;
     }
+
+    // Interrupt retrieval of incomplete images to free network resources when
+    // navigating pages in a PWA. Opt for tiny dataURI image instead of empty
+    // src to prevent the viewer from detecting a load error.
+    const img = dev().assertElement(this.img_);
+    if (!img.complete) {
```

Comment:
Would removing the img from DOM also stop the network request? As it is, if `unlayoutCallback` is ever called then `layoutCallback` later (imagine a user swipes to the next doc in the SERP carousel and then comes back), we'll be displaying this gif image instead of the real one. If we remove the img (and set `this.img_ = null`), then `layoutCallback` will reinitialize the image.

Reply:
Good thought, but surprisingly it does not stop the network connection. How about this sequence (tested and works)?

```
const img = dev().assertElement(this.img_);
if (!img.complete) {
  img.src =
    'data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs=';
  img.parentNode.removeChild(img);
  this.img_ = null;
}
```

Reply:
SGTM. Note, there's a remove helper in `src/dom.js` that the linter will probably complain that you use.

Reply:
Thanks for the heads-up. I'll push a new commit.